web/src/engine/websites/MangaSail.ts

```diff
@@ -0,0 +1,57 @@
+import { Tags } from '../Tags';
+import icon from './MangaSail.webp';
+import { Chapter, DecoratableMangaScraper, type Manga,type MangaScraper, type Page } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchCSS, FetchRequest } from '../FetchProvider';
+
+function IsImage(page: string) {
+    const link = page.toLowerCase();
+    let result = link.endsWith('.jpg') || link.endsWith('.jpeg');
+    result = result || link.endsWith('.png') || link.endsWith('.bmp') || link.endsWith('.avif') || link.endsWith('.webp') || link.endsWith('.bmp');
+    return result;
+}
+
+@Common.MangaCSS(/^https?:\/\/www\.mangasail\.net\/content\//, 'div.main-content-inner h1.page-header')
+@Common.MangasMultiPageCSS('/directory?page={page}', 'table.directory_list tr td:first-of-type a')
+@Common.ImageDirect()
+
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('mangasail', `MangaSail`, 'https://www.mangasail.net', Tags.Language.English, Tags.Media.Manga, Tags.Media.Manhua, Tags.Media.Manhwa, Tags.Source.Aggregator);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const chapterslist = [];
+        for (let page = 0, run = true; run; page++) {
+            const chapters = await this.getChaptersFromPage(page, manga);
+            chapters.length > 0 && !this.EndsWith(chapterslist, chapters) ? chapterslist.push(...chapters) : run = false;
+        }
+        return chapterslist;
+
+    }
+
+    private async getChaptersFromPage(page: number, manga: Manga): Promise<Chapter[]> {
+        const url = new URL(manga.Identifier + '?page=' + page, this.URI).href;
+        const request = new FetchRequest(url);
+        const data = await FetchCSS<HTMLAnchorElement>(request, 'table.chlist tr td:first-of-type a');
+        return data.map(anchor => new Chapter(this, manga, anchor.pathname + '?page=all', anchor.text.replace(manga.Title, '').trim()));
+    }
+
+    public override async FetchPages(this: MangaScraper, chapter: Chapter): Promise<Page[]> {
+        const data = await Common.FetchPagesSinglePageJS.call(this, chapter, 'Drupal.settings.showmanga.paths');
```

Comment:
The variable `data` seems to be of any type, is it possible to use correct type (is `v` generic)?